### ISSUE TITLE
Compile error in Screen.cpp line 27 - no matching function CFastLED::addLeds

### ISSUE THREAD

**Issue by cRoCx** [raidlman/1d-pong#4]
I've setup Arduino IDE and tried to compile this project without an Arduino connected just yet. The following error message pops up in [Screen.cpp#27](https://github.com/raidlman/1d-pong/blob/7a03ea0de621e74a311400ab0a429472601f9a95/Screen.cpp#L27).
Content of Screen.cpp line 27: `FastLED.addLeds<APA102, BGR>(leds, num_leds).setCorrection( TypicalLEDStrip );`

Error message:
```
Arduino: 1.8.13 (Windows 10), Board: "Adafruit Circuit Playground Bluefruit, 0.3.2 SoftDevice s140 6.1.1, Level 0 (Release)"
In file included from sketch\Player.h:28,
                 from sketch\Ball.h:28,
                 from sketch\Pong.h:30,
                 from sketch\Pong.cpp:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.003.003
   14 | #    pragma message "FastLED version 3.003.003"
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from sketch\Player.h:28,
                 from sketch\helper.cpp:3:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.003.003
   14 | #    pragma message "FastLED version 3.003.003"
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from sketch\Player.h:28,
                 from sketch\Ball.h:28,
                 from sketch\Ball.cpp:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.003.003
   14 | #    pragma message "FastLED version 3.003.003"
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from sketch\Player.h:28,
                 from sketch\Player.cpp:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.003.003
   14 | #    pragma message "FastLED version 3.003.003"
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from sketch\Player.h:28,
                 from sketch\Ball.h:28,
                 from sketch\Pong.h:30,
                 from D:\Sonstiges\Arduino\1dpong\1dpong.ino:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.003.003
   14 | #    pragma message "FastLED version 3.003.003"
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from sketch\Screen.h:28,
                 from sketch\Screen.cpp:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.003.003
   14 | #    pragma message "FastLED version 3.003.003"
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
sketch\Screen.cpp: In member function 'void Screen::init()':
Screen.cpp:27:46: error: no matching function for call to 'CFastLED::addLeds<APA102, BGR>(CRGB*&, uint8_t&)'
   27 |   FastLED.addLeds<APA102, BGR>(leds, num_leds).setCorrection( TypicalLEDStrip );
      |                                              ^
In file included from sketch\Screen.h:28,
                 from sketch\Screen.cpp:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:223:130: note: candidate: 'template<ESPIChipsets CHIPSET, unsigned char DATA_PIN, unsigned char CLOCK_PIN, EOrder RGB_ORDER, long unsigned int SPI_DATA_RATE> CLEDController& CFastLED::addLeds(CRGB*, int, int)'
  223 |  template<ESPIChipsets CHIPSET,  uint8_t DATA_PIN, uint8_t CLOCK_PIN, EOrder RGB_ORDER, uint32_t SPI_DATA_RATE > CLEDController &addLeds(struct CRGB *data, int nLedsOrOffset, int nLedsIfOffset = 0) {
      |                                                                                                                                  ^~~~~~~
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:223:130: note:   template argument deduction/substitution failed:
sketch\Screen.cpp:27:46: note:   couldn't deduce template parameter 'CLOCK_PIN'
   27 |   FastLED.addLeds<APA102, BGR>(leds, num_leds).setCorrection( TypicalLEDStrip );
      |                                              ^
In file included from sketch\Screen.h:28,
                 from sketch\Screen.cpp:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:237:95: note: candidate: 'template<ESPIChipsets CHIPSET, unsigned char DATA_PIN, unsigned char CLOCK_PIN> static CLEDController& CFastLED::addLeds(CRGB*, int, int)'
  237 |  template<ESPIChipsets CHIPSET,  uint8_t DATA_PIN, uint8_t CLOCK_PIN > static CLEDController &addLeds(struct CRGB *data, int nLedsOrOffset, int nLedsIfOffset = 0) {
      |                                                                                               ^~~~~~~
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:237:95: note:   template argument deduction/substitution failed:
sketch\Screen.cpp:27:46: note:   couldn't deduce template parameter 'CLOCK_PIN'
   27 |   FastLED.addLeds<APA102, BGR>(leds, num_leds).setCorrection( TypicalLEDStrip );
      |                                              ^
In file included from sketch\Screen.h:28,
                 from sketch\Screen.cpp:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:251:113: note: candidate: 'template<ESPIChipsets CHIPSET, unsigned char DATA_PIN, unsigned char CLOCK_PIN, EOrder RGB_ORDER> static CLEDController& CFastLED::addLeds(CRGB*, int, int)'
  251 |  template<ESPIChipsets CHIPSET,  uint8_t DATA_PIN, uint8_t CLOCK_PIN, EOrder RGB_ORDER > static CLEDController &addLeds(struct CRGB *data, int nLedsOrOffset, int nLedsIfOffset = 0) {
      |                                                                                                                 ^~~~~~~
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:251:113: note:   template argument deduction/substitution failed:
sketch\Screen.cpp:27:46: note:   couldn't deduce template parameter 'CLOCK_PIN'
   27 |   FastLED.addLeds<APA102, BGR>(leds, num_leds).setCorrection( TypicalLEDStrip );
      |                                              ^
In file included from sketch\Screen.h:28,
                 from sketch\Screen.cpp:20:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:301:25: note: candidate: 'template<template<unsigned char DATA_PIN, EOrder RGB_ORDER> class CHIPSET, unsigned char DATA_PIN, EOrder RGB_ORDER> static CLEDController& CFastLED::addLeds(CRGB*, int, int)'
  301 |  static CLEDController &addLeds(struct CRGB *data, int nLedsOrOffset, int nLedsIfOffset = 0) {
      |                         ^~~~~~~
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:301:25: note:   template argument deduction/substitution failed:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:307:25: note: candidate: 'template<template<unsigned char DATA_PIN, EOrder RGB_ORDER> class CHIPSET, unsigned char DATA_PIN> static CLEDController& CFastLED::addLeds(CRGB*, int, int)'
  307 |  static CLEDController &addLeds(struct CRGB *data, int nLedsOrOffset, int nLedsIfOffset = 0) {
      |                         ^~~~~~~
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:307:25: note:   template argument deduction/substitution failed:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:313:25: note: candidate: 'template<template<unsigned char DATA_PIN> class CHIPSET, unsigned char DATA_PIN> static CLEDController& CFastLED::addLeds(CRGB*, int, int)'
  313 |  static CLEDController &addLeds(struct CRGB *data, int nLedsOrOffset, int nLedsIfOffset = 0) {
      |                         ^~~~~~~
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:313:25: note:   template argument deduction/substitution failed:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:357:25: note: candidate: 'template<template<EOrder RGB_ORDER> class CHIPSET, EOrder RGB_ORDER> static CLEDController& CFastLED::addLeds(CRGB*, int, int)'
  357 |  static CLEDController &addLeds(struct CRGB *data, int nLedsOrOffset, int nLedsIfOffset = 0) {
      |                         ^~~~~~~
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:357:25: note:   template argument deduction/substitution failed:
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:363:25: note: candidate: 'template<template<EOrder RGB_ORDER> class CHIPSET> static CLEDController& CFastLED::addLeds(CRGB*, int, int)'
  363 |  static CLEDController &addLeds(struct CRGB *data, int nLedsOrOffset, int nLedsIfOffset = 0) {
      |                         ^~~~~~~
D:\Sonstiges\Arduino\libraries\FastLED-master\src/FastLED.h:363:25: note:   template argument deduction/substitution failed:
Screen.cpp:27:46: error: wrong number of template arguments (2, should be 1)
   27 |   FastLED.addLeds<APA102, BGR>(leds, num_leds).setCorrection( TypicalLEDStrip );
      |                                              ^
exit status 1
no matching function for call to 'CFastLED::addLeds<APA102, BGR>(CRGB*&, uint8_t&)'
```
Any idea what I did wrong?

**Comment by raidlman** [comment]
Did you install the FastLED version noted in README.md? It's just a guess but maybe the interface is not compatible with earlier/later versions.

**Comment by cRoCx** [comment]
You were absolutely right, thanks for the hint. I've got it to compile successfully for your board and that particular version of FastLED.

Now that I've got a different board (Adafruit Circuit Playground Bluefruit) I also got it up and running for it, also with the most recent version of FastLED. I've some more issues left to fix, but at least some LEDs are lighting up. Looks promising. Think I'll create a pull request for board support of my board sooner or later, once I'm done.

**Comment by cRoCx** [comment]
I've abandoned trying to get this code running on an Adafruit Circuit Playground Bluefruit.
The error message above occurs due to incompatibilities between the FastLED library and this board. The FastLED developers state it is compatible, but their own testing code fails to compile in recent versions of ArduinoIDE.

I've bought an Adafruit Pro Trinket 5V 16MHz instead and got the game running using the following commit (thx @raidlman for the hint):
https://github.com/raidlman/1d-pong/tree/729da8867d3ddb48265d4f743e9b3ef6d12e23c6

Tested it with FastLED v3.4.0.
I had to change the Pins, since Pin 7 is not available on the Adafruit Pro Trinket, but Pins 3-6, Pin 8 and Pins A0-A5 are.
